### PR TITLE
Delivery service

### PR DESCRIPTION
**Pull request by Nalivayko13**

### CODE REVIEW

File: controller/DeliveryServiceController.go [hunk 1]
```diff
@@ -40,3 +41,80 @@ func (h *Handler) CreateDeliveryService(ctx *gin.Context) {
...
+// @Produce json
+// @Success 200 {object} listDeliveryServices
+// @Failure 400 {string} string
+// @Failure 500 {string} string
+// @Router /deliveryservice [get]
+func (h *Handler) GetAllDeliveryServices(ctx *gin.Context) {
+	services, err := h.services.DeliveryServiceApp.GetAllDeliveryServices()
+	if err != nil {
+		ctx.JSON(http.StatusBadRequest, gin.H{"message": fmt.Sprintf("Error: %s", err)})
+		return
```
Comment (Baraulia) [inline controller/DeliveryServiceController.go:84]: You can't use status Bad Request  in this case (here server error)

File: dao/DeliveryServiceRepository.go [hunk 2]
```diff
@@ -26,11 +26,28 @@ type DeliveryService struct {
...
+	res, err := r.db.Query(fmt.Sprintf("SELECT id, name,email,photo,description,phone_number,manager_id,status FROM delivery_service Where id=%d", Id))
+	if err != nil {
+		log.Println(err)
+		return DeliveryService{}, err
+	}
+	for res.Next() {
+		err = res.Scan(&service.Id, &service.Name, &service.Email, &service.Photo, &service.Description, &service.PhoneNumber, &service.ManagerId, &service.Status)
+		if err != nil {
+			log.Println(err)
+			return DeliveryService{}, err
```
Comment (TonySinero) [inline dao/DeliveryServiceRepository.go]: I don't know if this is possible now, but it might be better to return nill

**Comment by Ann40a** [comment]
I've updated base branch for this PR to FDS-149. Because otherwise I've seen data from both tickets there. You would need to fix and merge FDS-149 first and only then merge this PR

File: dao/DeliveryServiceRepository.go [hunk 3]
```diff
@@ -26,11 +26,96 @@ type DeliveryService struct {
 }
 
 func (r *DeliveryServicePostgres) SaveDeliveryServiceInDB(service *DeliveryService) (int, error) {
-	row := r.db.QueryRow("INSERT INTO delivery_service (name, email, photo, description, phone_number, status) VALUES ($1, $2, $3, $4, $5, $6) RETURNING id", service.Name, service.Email, service.Photo, service.Description, service.PhoneNumber, service.Status)
+	row := r.db.QueryRow("INSERT INTO delivery_service (name, email, photo, description, phone_number,manager_id, status) VALUES ($1, $2, $3, $4, $5,$6, $7) RETURNING id", service.Name, service.Email, service.Photo, service.Description, service.PhoneNumber, service.ManagerId, service.Status)
```
Comment (Ann40a) [inline dao/DeliveryServiceRepository.go]:
```suggestion
	row := r.db.QueryRow("INSERT INTO delivery_service (name, email, photo, description, phone_number,manager_id, status) VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING id", service.Name, service.Email, service.Photo, service.Description, service.PhoneNumber, service.ManagerId, service.Status)
```

File: service/DeliveryServiceModel.go [hunk 4]
```diff
@@ -23,3 +24,32 @@ func (s *DeliveryService) CreateDeliveryService(DeliveryService dao.DeliveryServ
 	}
 	return id, nil
 }
+
+func (s *DeliveryService) GetDeliveryServiceById(Id int) (dao.DeliveryService, error) {
+	var service *dao.DeliveryService
+	service, err := s.repo.GetDeliveryServiceByIdFromDB(Id)
+	if err != nil {
+		err := errors.New("no id")
+		log.Println("no more id")
```
Comment (Ann40a) [inline service/DeliveryServiceModel.go]: not found

File: tests/DeliveryServiceController_test.go [hunk 5]
```diff
@@ -77,3 +77,118 @@ func TestHandler_CreateDeliveryService(t *testing.T) {
...
+			r.ServeHTTP(w, req)
+
+			assert.Equal(t, tt.expectedStatusCode, w.Code)
+			assert.Equal(t, tt.expectedRequestBody, w.Body.String())
+
+		})
+	}
+}
+
+func TestHandler_GetDeliveryServiceById(t *testing.T) {
```
Comment (Ann40a) [inline tests/DeliveryServiceController_test.go:137]: let's also have test for update

**Comment by Ann40a** [comment]
Returning the ticket to in-progress because it depends on FDS-149 and has some minor errors.

**Pull request closed without merging**